shared/Microsoft.Extensions.ActivatorUtilities.Sources/ActivatorUtilities.cs

```diff
@@ -294,7 +294,7 @@ public object CreateInstance(IServiceProvider provider)
                 {
                     return _constructor.Invoke(_parameterValues);
                 }
-                catch (Exception ex)
+                catch (Exception ex) when (ex.InnerException != null)
```

Comment:
Yeah, that would work as well. There are some other exceptions that `Invoke` _might_ throw though. Do you think it's safe to assume it will just throw a `TargetInvocationException`?